src/worker.rs

```diff
@@ -218,6 +225,8 @@ impl WorkerPool {
 
     pub fn wait(&mut self) {
         self.notify.clear();
+        self.wstate.exit.store(true, Ordering::SeqCst);
```

Comment:
In my experiments this didn't see to be fully working. I never got a chance to check why

Reply:
What were the symptoms of it failing / how can I reproduce a failing case?

Reply:
The worker would not actually stop, and that function would wait forever.